include/pcmki/pcmki_transition.h

```diff
@@ -28,19 +28,39 @@ typedef enum {
 typedef struct te_timer_s crm_action_timer_t;
 typedef struct crm_graph_s crm_graph_t;
 
+enum pcmk__synapse_flags {
+    pcmk__synapse_ready       = (1 << 0),
+    pcmk__synapse_failed      = (1 << 1),
+    pcmk__synapse_executed    = (1 << 2),
+    pcmk__synapse_confirmed   = (1 << 3),
+};
+
 typedef struct synapse_s {
     int id;
     int priority;
 
-    gboolean ready;
-    gboolean failed;
-    gboolean executed;
-    gboolean confirmed;
+    uint32_t flags; // Group of pcmk__synapse_flags
 
     GList *actions;           /* crm_action_t* */
     GList *inputs;            /* crm_action_t* */
 } synapse_t;
 
+const char *synapse_state_str(synapse_t *synapse);
+
+#define pcmk__set_synapse_flags(synapse, flags_to_set) do {             \
+        (synapse)->flags = pcmk__set_flags_as(__func__, __LINE__,       \
+            LOG_TRACE,                                                  \
+            "Synapse", synapse_state_str(synapse),                       \
```

Comment:
I just noticed you're using synapse_state_str() as the "target" argument. This is used as the "for ..." in the log message, e.g.:

    Synapse flags 0x00000100 (pcmk__synapse_confirmed) for ... set by pcmk_sched_transition.c:599

synapse_state_str() doesn't really help here ("for Pending", "for Complete", etc.). Ideally, we'd use synapse->id like:

    Synapse flags 0x00000100 (pcmk__synapse_confirmed) for synapse 2 set by pcmk_sched_transition.c:599

Unfortunately id is an integer and pcmk__set_flags_as() needs a string. It's not worth the function call and memory allocation to create a "synapse %d" string just to use in trace messages, so I'm thinking we should just use the literal string "synapse" for the target. The message will look like

    Synapse flags 0x00000100 (pcmk__synapse_confirmed) for synapse set by pcmk_sched_transition.c:599

It doesn't add anything useful to the log but it's the best we can do without unnecessary overhead.